aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterQueryTest.java

```diff
@@ -449,6 +449,39 @@ public void queryWithOrderByRelatedModel() throws DataStoreException {
         assertEquals(sorted, result);
     }
 
+    /**
+     * Test that new QueryField with explicit model name produces the same result as old QueryField.
+     * @throws DataStoreException On failure to arrange items into store, or from the query action itself
+     */
+    @Test
+    public void queryFieldsAreBackwardsCompatible() throws DataStoreException {
+        BlogOwner blogOwner = BlogOwner.builder().name("Test Dummy").build();
+        adapter.save(blogOwner);
+        Blog blog = Blog.builder().name("Blogging for Dummies").owner(blogOwner).build();
+        adapter.save(blog);
+
+        final int numModels = 10;
+        for (int counter = 0; counter < numModels; counter++) {
+            final Post post = Post.builder()
+                    .title("title " + counter)
+                    .status(PostStatus.INACTIVE)
+                    .rating(counter)
+                    .blog(blog)
+                    .build();
+            adapter.save(post);
+        }
+
+        // Assert that using QueryField without model name yields same results if there is no column ambiguity
+        assertEquals(
+            adapter.query(Post.class, Where.matches(field("Post", "title").contains("4"))),
+            adapter.query(Post.class, Where.matches(field("title").contains("4")))
+        );
```

Comment:
There's probably some cool way to start from just the query fields, and then build towards an `allSame()` predicate. I've got a start here but this still isn't exactly it....

```java
Observable.fromArray(field("Post", 'title), field("title))
    .map(queryField -> Where.matches(queryField).contains("4"))
    .toList()
    .map(HashSet::new)
    .awaitCount(1)
    .assertValue(set -> set.size() == 0);
```